os/drivers/bluetooth/virtual_ble/vble_handler.c

```diff
@@ -0,0 +1,472 @@
+                 sleep_time, result, event_type.evt);
+
+    sleep(sleep_time);
+    vble_callback_handler(event_type, result);
+}
+
+int vble_create_event(uint32_t event, int32_t result, int32_t sleep)
+{
+    char event_buf[16] = {0,};
+    char res_buf[16] = {0,};
+    char sleep_buf[16] = {0,};
+
+    memset(event_buf, 0, 16);
+    memset(res_buf, 0, 16);
+    memset(sleep_buf, 0, 16);
+
+    snprintf(event_buf, 16, "%d", event);
+    snprintf(res_buf, 16, "%d", result);
+    snprintf(sleep_buf, 16, "%d", sleep);
+
+    char *argv[4] = {0,};
+    argv[0] = event_buf;
+    argv[1] = res_buf;
+    argv[2] = sleep_buf;
+    argv[3] = NULL;
+    int res = kernel_thread("vble_evt", 100, 2048, _generate_evt, argv);
+    if (res == -1) {
+        VBLE_ERROR(0);
+        return -1;
+    }
+    return 0;
+}
+
+static void init_handler(struct vconfigs *config) {
+    //init client
+    client_init_parm.trble_scan_state_changed_cb = config->client->trble_scan_state_changed_cb;
+    client_init_parm.trble_device_scanned_cb = config->client->trble_device_scanned_cb;
+    client_init_parm.trble_device_connected_cb = config->client->trble_device_connected_cb;
+    client_init_parm.trble_device_disconnected_cb = config->client->trble_device_disconnected_cb;
+    client_init_parm.trble_operation_notification_cb = config->client->trble_operation_notification_cb;
+    client_init_parm.mtu = config->client->mtu;
+
+    //init server
+    server_profile_count = config->server->profile_count;
+    uint16_t gatt_char_num = 0;
+    for (int i = 0; i < config->server->profile_count; i++)
+    {
+        if (config->server->profile[i].type == TRBLE_GATT_CHARACT) {
+            gatt_char_num++;
+        }
+    }
+    server_init_parm.profile_count = config->server->profile_count + gatt_char_num;
+    server_init_parm.profile = config->server->profile;
+    server_init_parm.connected_cb = config->server->connected_cb;
+    server_init_parm.is_secured_connect_allowed = config->server->is_secured_connect_allowed;
+    is_server_init = true;
+    is_client_init = true;
+    for (uint16_t i = 0; i < TRBLE_MAX_CONNECTION_COUNT; i++)
+    {
+        vble_connected_list.conn_handle[i] = VBLE_NO_CONNECTION;
+    }
+    vble_connected_list.connected_count = 0;
+    attr_table.count = 0;
+    return;
+}
+
+static void connect_handler(trble_conn_info *conn_info) {
+    vble_connect.conn_info = *conn_info;
+    vdev_reverse_mac(vble_connect.conn_info.addr.mac);
+    uint8_t index = vble_connected_list.connected_count;
+    connected_device[index].conn_info = *conn_info;
+    connected_device[index].is_bonded = conn_info->is_secured_connect;
+    connected_device[index].conn_handle = index;
+    vble_connected_list.conn_handle[index] = index;
+    return;
+}
+
+static void disconnect_all_handler(void) {
+    vble_connected_list.connected_count = 0;
+    bool check = false;
+    for (uint16_t i = 0; i < TRBLE_MAX_CONNECTION_COUNT; i++)
```

Comment:
Is there any special reason to use `uint16_t`? 
`i` is just an index, so I don't know we need unsigned.

Reply:
Done